### PR TITLE
auto case OCP-33206

### PR DESCRIPTION
**Pull request by zhaozhanqi**
@anuragthehatter @weliang1 @huiran0826 @rbbratta  PTAL, thanks
logs:
http://pastebin.test.redhat.com/888205

### CODE REVIEW

**Comment by zhaozhanqi** [comment]
Just saw OVN also have this feature.  maybe I need to update this for OVN

File: features/networking/sdn.feature [hunk 1]
```diff
@@ -392,3 +392,30 @@ Feature: SDN related networking scenarios
...
+    And I use the "openshift-monitoring" project
+    When evaluation of `secret(service_account('prometheus-k8s').get_secret_names.find {|s| s.match('token')}).token` is stored in the :sa_token clipboard
+    When I run the :exec admin command with:
+      | n                | openshift-monitoring |
+      | pod              | prometheus-k8s-0     |
+      | c                | prometheus           |
+      | oc_opts_end      |                      |
+      | exec_command     | sh                   |
+      | exec_command_arg | -c                   |
+      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://<%= cb.sdn_metri_endpoint %>/metrics |
```
Comment (rbbratta) [inline features/networking/sdn.feature:418]: do we need to use `sh -c` in this case?

Comment (anuragthehatter) [reply features/networking/sdn.feature:418]: I guess if ```sh -c``` or ```bash -c``` is not specified for containers , the rest of the keywords after curl would be assumed as curl arguments which would fatal the run. Using sh/bash would tie whole command as per my experience. Not sure whats the latest is

**Comment by anuragthehatter** [comment]
> Just saw OVN also have this feature. maybe I need to update this for OVN

Yea, but might be complicated though to make this case compatible to both. We don't have metrics cases for OVN though. You can search metrics staring with ```ovnkube-master``` or ```ovnkube-node``` though

File: features/networking/sdn.feature (hunk 1 above)
Comment (zhaozhanqi) [reply features/networking/sdn.feature:418]: yes. As @anuragthehatter said

File: features/networking/sdn.feature [hunk 2]
```diff
@@ -392,3 +392,30 @@ Feature: SDN related networking scenarios
...
       | Not applying unsafe change: invalid configuration |
+
+
+  # @author zzhao@redhat.com
+  # @case_id OCP-33206
+  @admin
+  Scenario: sdn metrics should be using https scheme
+    Given I switch to cluster admin pseudo user
+    When I use the "openshift-sdn" project
+    Then the expression should be true> service_monitor('monitor-sdn').service_monitor_endpoints_spec.first.scheme == 'https'
```
Comment (pruan-rht) [inline features/networking/sdn.feature:403]: @zhaozhanqi @anuragthehatter pointed out in #1461 that the endpoint names are different for OCP < 4.4.  So if you want this testcase to work for all version of OCP, we need to change the logic a bit.

Comment (anuragthehatter) [reply features/networking/sdn.feature:403]: @pruan-rht This is a specific 4.6 feature so no need to implement that logic here. Right @zhaozhanqi ?

Comment (pruan-rht) [reply features/networking/sdn.feature:403]: Then perhaps add a filter at the beginning to make sure the ocp version is >= 4.6

Comment (zhaozhanqi) [reply features/networking/sdn.feature:404]: yes, this case need version >=4.6.  I added the step >=4.6
Actually. this case cannot be selected for <4.6 by case filter.

File: features/networking/sdn.feature (hunk 1 above)
Comment (rbbratta) [reply features/networking/sdn.feature:419]: the regular table form works in my testing

```cucumber
    And the output should contain "Hello OpenShift"
      | n                | openshift-monitoring                         |
      | pod              | prometheus-k8s-0                             |
      | c                | prometheus                                   |
      | oc_opts_end      |                                              |
      | exec_command     | curl                                         |
      | exec_command_arg | -k                                           |
      | exec_command_arg | -H                                           |
      | exec_command_arg | Authorization: Bearer <%= cb.sa_token %>     |
      | exec_command_arg | https://<%= cb.sdn_metri_endpoint %>/metrics |
```

Comment (anuragthehatter) [reply features/networking/sdn.feature:419]: That's seems right. Yeah, passing each flag and its argument as separate under ```exec_command_arg```.  Cool find. Nice!

Comment (anuragthehatter) [reply features/networking/sdn.feature:419]: But it seems to be required in such cases where where things are ```; or :``` separated

Comment (anuragthehatter) [reply features/networking/sdn.feature:419]: I guess to avoid spending time in figurng out what works or not its better we stick to ```bash -c```

Comment (rbbratta) [reply features/networking/sdn.feature:419]: exec only works for direct single command lines as it does use the shell at all.

These are the bash metacharacters:

```shell
       metacharacter
              A character that, when unquoted, separates words.  One of the following:
              |  & ; ( ) < > space tab newline
       control operator
              A token that performs a control function.  It is one of the following symbols:
              || & && ; ;; ;& ;;& ( ) | |& <newline>
```

and any other shell keyword:

```shell
       ! case  coproc  do done elif else esac fi for function if in select then until while { } time [[ ]]
```
Any of these require using `bash`


Once I was bored and read the entire `bash` man page, it is very useful, tons of cool tricks. :smiley:

**Comment by zhaozhanqi** [comment]
I think this PR can be closed since another PR #1461 already cover this, thanks

**Pull request closed without merging**